engine/src/ast/function_expr.rs

```diff
@@ -312,149 +398,212 @@ fn test_function() {
+                                        SCHEME
+                                            .get_field_index("http.request.headers.is_empty")
+                                            .unwrap()
+                                    ),
+                                    indexes: vec![],
+                                },
+                                op: FieldOp::IsTrue,
+                            }))
                         ]
+                    })
+                ))],
+            },
+            ""
+        );
+
+        assert_json!(
+            expr,
+            {
+                "name": "any",
+                "args": [
+                    {
+                        "kind": "SimpleExpr",
+                        "value": {
+                            "items": [
+                                {
+                                    "lhs": "http.request.headers.is_empty",
+                                    "op": "IsTrue",
+                                },
+                                {
+                                    "lhs": "http.request.headers.is_empty",
+                                    "op": "IsTrue",
+                                }
+                            ],
+                            "op": "Or",
+                        }
                     }
-                }
-            ]
-        }
-    );
-
-    assert_err!(
-        FunctionCallExpr::lex_with("echo ( \"test\" );", &SCHEME),
-        LexErrorKind::InvalidArgumentKind {
-            index: 0,
-            mismatch: FunctionArgKindMismatchError {
-                actual: FunctionArgKind::Literal,
-                expected: FunctionArgKind::Field,
+                ]
             }
-        },
-        "\"test\""
-    );
-
-    assert_err!(
-        FunctionCallExpr::lex_with("echo ( 10 );", &SCHEME),
-        LexErrorKind::InvalidArgumentKind {
-            index: 0,
-            mismatch: FunctionArgKindMismatchError {
-                actual: FunctionArgKind::Literal,
-                expected: FunctionArgKind::Field,
-            }
-        },
-        "10"
-    );
-
-    assert_err!(
-        FunctionCallExpr::lex_with("echo ( ip.addr );", &SCHEME),
-        LexErrorKind::InvalidArgumentType {
-            index: 0,
-            mismatch: TypeMismatchError {
-                actual: Type::Ip,
-                expected: Type::Bytes.into(),
-            }
-        },
-        "ip.addr"
-    );
-
-    assert_err!(
-        FunctionCallExpr::lex_with("echo ( http.host, 10, \"test\" );", &SCHEME),
-        LexErrorKind::InvalidArgumentsCount {
-            expected_min: 1,
-            expected_max: Some(2),
-        },
-        "\"test\" );"
-    );
-
-    assert_err!(
-        FunctionCallExpr::lex_with("echo ( http.test );", &SCHEME),
-        LexErrorKind::UnknownField(UnknownFieldError),
-        "http.test"
-    );
-
-    assert_err!(
-        FunctionCallExpr::lex_with("echo ( echo ( http.test ) );", &SCHEME),
```

Comment:
Why removing this test? It was added specifically for testing the error reporting.

Reply:
It wasn't intentional, was deleted by accident during rebase.

Reply:
It wasn't deleted, we still have this test, but the test now assert_ok! because this is a valid function call.

Reply:
It should not be a valid function call because http.test does not exists. This test was added to test error reporting for nested function call.